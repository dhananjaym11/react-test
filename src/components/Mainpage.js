import React, { Component } from 'react';

import CardList from './CardList';
import SearchBox from './SearchBox';
import Scroll from './Scroll';
import ErrorBoundry from './ErrorBoundry';
import Header from './Header';

class Mainpage extends Component {

    componentDidMount() {
        this.props.onRequestRobots();
    }

    getFilteredRobots = () => {
        return this.props.robots.filter(robot => {
            return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
        })
    }

    render() {
        const { onSearchChange, isPending } = this.props;
        return (
            <div className='tc'>
                <Header />
                <SearchBox searchChange={onSearchChange} />
                <Scroll>
                    {isPending ? <h1>Loading</h1> :
                        <ErrorBoundry>
                            <CardList robots={this.getFilteredRobots()} />
                        </ErrorBoundry>
                    }
                </Scroll>
            </div>
        );
    }
}

export default Mainpage;
