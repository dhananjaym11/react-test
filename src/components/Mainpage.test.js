import { shallow } from 'enzyme';
import React from 'react';
import Mainpage from './Mainpage';

let wrapper;
beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [
            { id: 1, name: 'aa', email: 'aa@gmail.com' },
            { id: 2, name: 'bb', email: 'bb@gmail.com' }
        ],
        searchField: 'a',
        isPending: false
    }

    wrapper = shallow(<Mainpage {...mockProps} />)
})

it('Render component', () => {
    expect(wrapper).toMatchSnapshot();
})

it('Filter correctly', () => {
    expect(wrapper.instance().getFilteredRobots()).toEqual([{ id: 1, name: 'aa', email: 'aa@gmail.com' }]);
})