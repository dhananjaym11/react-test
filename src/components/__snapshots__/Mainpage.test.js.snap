// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Mainpage
    isPending={false}
    onRequestRobots={[Function]}
    robots={
        Array [
            Object {
              "email": "aa@gmail.com",
              "id": 1,
              "name": "aa",
            },
            Object {
              "email": "bb@gmail.com",
              "id": 2,
              "name": "bb",
            },
          ]
    }
    searchField="a"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <SearchBox
          searchChange={undefined}
/>,
        <Scroll>
          <ErrorBoundary>
                    <CardList
                              robots={
                                        Array [
                                                  Object {
                                                    "email": "aa@gmail.com",
                                                    "id": 1,
                                                    "name": "aa",
                                                  },
                                                ]
                              }
                    />
          </ErrorBoundary>
</Scroll>,
      ],
      "className": "tc",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "searchChange": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ErrorBoundary>
            <CardList
                        robots={
                                    Array [
                                                Object {
                                                  "email": "aa@gmail.com",
                                                  "id": 1,
                                                  "name": "aa",
                                                },
                                              ]
                        }
            />
</ErrorBoundary>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <CardList
              robots={
                            Array [
                                          Object {
                                            "email": "aa@gmail.com",
                                            "id": 1,
                                            "name": "aa",
                                          },
                                        ]
              }
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "robots": Array [
                Object {
                  "email": "aa@gmail.com",
                  "id": 1,
                  "name": "aa",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <SearchBox
            searchChange={undefined}
/>,
          <Scroll>
            <ErrorBoundary>
                        <CardList
                                    robots={
                                                Array [
                                                            Object {
                                                              "email": "aa@gmail.com",
                                                              "id": 1,
                                                              "name": "aa",
                                                            },
                                                          ]
                                    }
                        />
            </ErrorBoundary>
</Scroll>,
        ],
        "className": "tc",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "searchChange": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ErrorBoundary>
              <CardList
                            robots={
                                          Array [
                                                        Object {
                                                          "email": "aa@gmail.com",
                                                          "id": 1,
                                                          "name": "aa",
                                                        },
                                                      ]
                            }
              />
</ErrorBoundary>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <CardList
                robots={
                                Array [
                                                Object {
                                                  "email": "aa@gmail.com",
                                                  "id": 1,
                                                  "name": "aa",
                                                },
                                              ]
                }
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "robots": Array [
                  Object {
                    "email": "aa@gmail.com",
                    "id": 1,
                    "name": "aa",
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
